### PR TITLE
Configure Discovery Server using names [12349]

### PR DESCRIPTION
**Pull request by EduPonz**
This PR allows configuring both Discovery Servers and Clients specifying names instead of IPv4 addresses (port is still required). If a name is specified, the DNS is queried for IPv4 addresses, which are in turn used to configure the Server and Client.

* [x] Implementation (this PR)
* [x] System tests -> eProsima/Discovery-Server#38
* [x] Documentation -> eProsima/Fast-DDS-docs#282

### CODE REVIEW

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
Windows is issuing a warning due to the `getenv` method that has to be removed. However, there are two tests related to this PR that are failing in Windows:

* `LocatorDNSTests.resolve_name`
* `LocatorDNSTests.dns_locator`

**Comment by EduPonz** [comment]
> * `LocatorDNSTests.resolve_name`
> * `LocatorDNSTests.dns_locator`

This tests have failed in [today's nightly](http://jenkins.eprosima.com:8080/job/nightly_fastdds_sec_master_linux/771/), even though they did not failed in #2061. This is related to the known hosts in the testing machines, and has anything to do with the tests of the implementation.

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM if the CI does not give any unexpected failure.

**Comment by EduPonz** [comment]
@richiprosima please test windows

File: tools/fds/README.txt [hunk 1]
```diff
@@ -41,3 +42,8 @@ Examples:
 
       $ fast-discovery-server -i 1 -l 172.30.144.1 -p 12345 -b
```
Comment (JLBuenoLopez-eProsima) [inline tools/fds/README.txt:43]: I know these are previous typos but I think we should fix them:

```suggestion
      $ fast-discovery-server -i 4 -l 172.30.144.1 -p 12345 -b
```

Also, the previous example is also mistaken, both the server ID as the first UDP port are not correct:

```
      $ fast-discovery-server -i 3 -l 192.168.36.34 -p 8783 -l 172.20.96.1 -p 51083
```

File: tools/fds/server.h [hunk 2]
```diff
@@ -89,7 +90,12 @@ const option::Descriptor usage[] = {
       "\t   listening on 172.30.144.1 with UDP port 12345 and provided with a\n"
       "\t   backup file. If the server crashes it will automatically restore its\n"
       "\t   previous state when reenacted.\n\n"
-      "\t$ " FAST_SERVER_BINARY " -i 1 -l 172.30.144.1 -p 12345 -b" },
+      "\t$ " FAST_SERVER_BINARY " -i 1 -l 172.30.144.1 -p 12345 -b\n\n"
```
Comment (JLBuenoLopez-eProsima) [inline tools/fds/server.h:93]:
```suggestion
      "\t$ " FAST_SERVER_BINARY " -i 4 -l 172.30.144.1 -p 12345 -b\n\n"
```

Also, the previous example has the same errors as in the `README`:

```
      "\t$ " FAST_SERVER_BINARY " -i 3 -l 192.168.36.34 -p 8783 -l 172.20.96.1 -p 51083\n\n"
```

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Not really from your PR but I suggest fixing these typos in this PR.

File: tools/fds/README.txt (hunk 1 above)
Comment (EduPonz) [reply tools/fds/README.txt:43]: Amazing catch!

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**